### ISSUE TITLE
boost_build missing attribute

### ISSUE THREAD

**Issue by doevelopper** [bazelbuild/rules_foreign_cc#1071]
Hello, I am trying to build a boots's component, and the examble below leads to an issue ...
```py
 boost_build(
     name = "coroutine",
     lib_source = ":all_content",
     out_static_libs = ["libboost_coroutine.a"],
     user_options = [
         "--with-coroutine",
        "-j`nproc`",
        "link=static",
        "runtime-link=static",
    ],
     visibility = ["//visibility:public"],
 )
```
```txt
    .../external/rules_foreign_cc/foreign_cc/private/framework.bzl", line 848, column 22, in _define_inputs
                for tool in attrs.tools_data:
Error: 'struct' value has no field or method 'tools_data'

Available attributes: additional_inputs, additional_tools, alwayslink, build_data,  .........
```
 None of the commands listed in this link [rules_foreign_cc boost_build](https://bazelbuild.github.io/rules_foreign_cc/main/flatten.html#boost_build)  describes how to set it

All other  API (cmake,messon,make...) implements  in exemple for **make** in  _...rules_foreign_cc/foreign_cc/make.bzl_
```py
    tools_data = [make_data]
```
and there is no equivalent for _.../rules_foreign_cc/foreign_cc/boost_build.bzl_

Any help or tip of how to proceed  to build my library will be very helpfull
Thanks a lot

**Comment by jsharpe** [comment]
In the first instance I'd suggest using the rules from the nelhage/rules_boost repo first if you can as these are native bazel rules.
Otherwise patches are very welcome against the boost_build rules to fix this up; its likely just missing from the attribute list for boost_build (which isn't very well supported TBH; there are no tests for it and I'm not really aware of anyone that uses it..)

**Comment by doevelopper** [comment]
Hello @jsharpe , thanks for your reply,
I am aware of  `nelhage/rules_boost`,  the fact is that it introduces third party library  version that collides with mines...  🤕 .
I will try to implement a  _**rules_foreign_cc**_  first.

**Comment by sangwookim1003** [comment]
I have same problem.
Any update on this issue?

**Comment by attilaolah** [comment]
I stumbled upon this problem too. Lemme try to figure out how to properly use `local_repository` with modules to try and provide a pull request.

**Comment by attilaolah** [comment]
Should be fixed by #1112.
